android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/components/TopScreenSection.kt

```diff
@@ -134,62 +143,68 @@ fun TopScreenSection(
-      SetupToolbarIcons(
-        menuIcons = topScreenSection?.menuIcons,
-        navController = navController,
-        modifier = modifier,
-        onClick = onClick,
-        decodeImage = decodeImage,
-      )
-
-      if (isFilterIconEnabled) Spacer(modifier = Modifier.width(24.dp))
+      Row(verticalAlignment = Alignment.CenterVertically) {
+        Icon(
+          when (toolBarHomeNavigation) {
+            ToolBarHomeNavigation.OPEN_DRAWER -> Icons.Filled.Menu
+            ToolBarHomeNavigation.NAVIGATE_BACK -> Icons.AutoMirrored.Filled.ArrowBack
+          },
+          contentDescription = DRAWER_MENU,
+          tint = Color.White,
+          modifier =
+            modifier
+              .clickable { onClick(ToolbarClickEvent.Navigate) }
+              .testTag(TOP_ROW_ICON_TEST_TAG),
+        )
+        Text(
+          text = title,
+          fontSize = 20.sp,
+          color = Color.White,
+          modifier = modifier.padding(start = 16.dp).testTag(TOP_ROW_TEXT_TEST_TAG),
+        )
+      }
 
-      if (isFilterIconEnabled) {
-        BadgedBox(
-          modifier = Modifier.padding(end = 8.dp),
-          badge = {
-            if (filteredRecordsCount != null && filteredRecordsCount > -1) {
-              Badge {
-                Text(
-                  text = if (filteredRecordsCount > 99) "99+" else filteredRecordsCount.toString(),
-                  overflow = TextOverflow.Clip,
-                  maxLines = 1,
-                )
+      Row(
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = Modifier.padding(start = 8.dp),
+      ) {
+        if (isFilterIconEnabled) {
+          BadgedBox(
+            modifier = Modifier.padding(end = 8.dp),
+            badge = {
+              if (filteredRecordsCount != null && filteredRecordsCount > -1) {
+                Badge {
+                  Text(
+                    text =
+                      if (filteredRecordsCount > 99) "99+" else filteredRecordsCount.toString(),
+                    overflow = TextOverflow.Clip,
+                    maxLines = 1,
+                  )
+                }
               }
-            }
-          },
-        ) {
-          Icon(
-            imageVector = Icons.Default.FilterAlt,
-            contentDescription = FILTER,
-            tint = Color.White,
-            modifier =
-              modifier
-                .clickable { onClick(ToolbarClickEvent.FilterData) }
-                .testTag(TOP_ROW_FILTER_ICON_TEST_TAG),
-          )
+            },
+          ) {
+            Icon(
+              imageVector = Icons.Outlined.FilterAlt,
+              contentDescription = FILTER,
+              tint = Color.White,
+              modifier =
+                modifier
+                  .clickable { onClick(ToolbarClickEvent.FilterData) }
+                  .testTag(TOP_ROW_FILTER_ICON_TEST_TAG),
+            )
+          }
         }
```

Comment:
@ellykits Is there a reason for rearranging the icon placement?  This makes the filter icon to always default as first on registers

Reply:
Yes, this is because I introduced an overflow menu at the end if there are more than two icons. This register filter implementation was implemented as a standalone, it's not activated via the `topScreenMenuIcons` actions.

Reply:
Can I suggest we use a different icon for the map filter to avoid this confusion?

Reply:
@HenryRae Can you help with this? Can we have an icon to represent filtering data on a map for instance placing the funnel icon on top of the map icon?

Reply:
can you share a screenshot? it's not clear to me why we'd want a different icon, since the action (filtering) is the same, and that's what we want the icon to represent

Reply:
@dubdabasoduba showed me a screen, he'll reach out to clarify